### PR TITLE
fix: APPS-2491 Component updates

### PR DESCRIPTION
**Pull request by tinuola**
Connected to [APPS-2491](https://jira.library.ucla.edu/browse/APPS-2491)

**Notes:**
- Removed props from BlockStaffSubjectLibrarian component
- Updated logic to display dismiss icon ('icon-close') in ButtonLink component
- Refactored lingering use of ::v-deep in components
- Addressed linter errors and associated cleanup
- Added App.vue to .gitignore per discussion with @pghorpade 

**Checklist:**

-   [X] I checked that it is working locally in the dev server
-   [X] I checked that it is working locally in the storybook
-   [X] I checked that it is working locally in the 
library-website-nuxt dev server
-   [ ] I added a screenshot of it working
-   [ ] UX has reviewed and approved this
-   [X] I assigned this PR to someone on the dev team to review
-   [X] I used a conventional commit message
-   [X] I assigned myself to this PR

[APPS-2491]: https://uclalibrary.atlassian.net/browse/APPS-2491?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pghorpade — APPROVED** [review]
💯 thanks for fixing the v-deep to deep

File: src/lib-components/ButtonLink.vue [hunk 1]
```diff
@@ -93,8 +93,9 @@ export default {
           return 'svg-arrow-download'
         case isInternalLink(this.to):
           return 'svg-arrow-right'
-        case this.iconName === 'none':
+        case this.iconName === '':
```
Comment (pghorpade) [inline src/lib-components/ButtonLink.vue:96]: before this add 
Add another case 
to check the linkTarget
case this.linkTarget === '_blank' || this.iconName === 'svg-external-link':
    return 'svg-external-link'

Comment (pghorpade) [reply src/lib-components/ButtonLink.vue:96]: remove 
case this.iconName === '':
 return 'none'
 
 make 
default:
return 'none'

**Review by pghorpade — CHANGES_REQUESTED** [review]

**Comment by tinuola** [comment]
@pghorpade Changes made. Tx.

**Review by pghorpade — APPROVED** [review]
👍  nice work.

**Pull request merged**